examples/notebooks/beam-ml/mltransform/compute_and_apply_vocab.ipynb

```diff
@@ -0,0 +1,326 @@
+      ],
+      "metadata": {
+        "id": "oZZtAVk4wJCd"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "with beam.Pipeline() as pipeline:\n",
+        "  data_pcoll = pipeline | \"CreateData\" >> beam.Create(documents)\n",
+        "  # Compute and apply vocabulary by using MLTransform.\n",
+        "  transformed_pcoll = (\n",
+        "      data_pcoll\n",
+        "      | \"MLTransform\" >> MLTransform(write_artifact_location=artifact_location_with_frequency_threshold).with_transform(\n",
+        "          ComputeAndApplyVocabulary(columns=['feature'], split_string_by_delimiter=' ', frequency_threshold=2))\n",
+        "      )\n",
+        "  transformed_pcoll | \"Print\" >> beam.Map(print)"
+      ],
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "FpDkOSWtv1pn",
+        "outputId": "6fe43447-ea39-43f7-d9e7-07d35e846bfc"
+      },
+      "execution_count": 20,
+      "outputs": [
+        {
+          "output_type": "stream",
+          "name": "stderr",
+          "text": [
+            "WARNING:absl:You are passing instance dicts and DatasetMetadata to TFT which will not provide optimal performance. Consider following the TFT guide to upgrade to the TFXIO format (Apache Arrow RecordBatch).\n",
+            "WARNING:absl:You are passing instance dicts and DatasetMetadata to TFT which will not provide optimal performance. Consider following the TFT guide to upgrade to the TFXIO format (Apache Arrow RecordBatch).\n",
+            "WARNING:absl:You are outputting instance dicts from `TransformDataset` which will not provide optimal performance. Consider setting  `output_record_batches=True` to upgrade to the TFXIO format (Apache Arrow RecordBatch). Encoding functionality in this module works with both formats.\n"
+          ]
+        },
+        {
+          "output_type": "stream",
+          "name": "stdout",
+          "text": [
+            "Row(feature=array([ 0,  1,  4,  3, -1, -1,  0, -1, -1]))\n",
+            "Row(feature=array([ 0, -1, -1, -1, -1, -1,  2,  0, -1]))\n",
+            "Row(feature=array([-1, -1, -1,  2,  0, -1]))\n",
+            "Row(feature=array([0, 1, 4, 3]))\n"
+          ]
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## `MLTransform` for inference workloads\n",
+        "\n",
+        "When `MLTransform` is in `write` mode, it produces artifacts, such as vocabulary files for `ComputeAndApplyVocabulary`. When `MLTransform` is used `read` mode, it uses the previously generated vocabulary files to map the incoming text data. If the incoming vocabulary isn't found in the generated vocabulary, then the incoming vocabulary is mapped to a `default_value` provided during the `write` mode. In this case, the `default_value` is `-1`."
```

Comment:
It would be good to have a brief callout of why this matters. Something like:

```
When `MLTransform` is in `write` mode, it produces artifacts, such as vocabulary files for `ComputeAndApplyVocabulary`.

This allows you to ensure that you are applying the same vocabulary (and any other preprocessing transforms you apply) when you are training your model and serving it in production or testing its accuracy.
```

Feel free to improve on the wording here